src/errors.rs

```diff
@@ -90,6 +90,9 @@ pub enum OciDistributionError {
     /// Versioned object: JSON deserialization error
     #[error("Failed to parse manifest: {0}")]
     VersionedParsingError(String),
+    #[error(transparent)]
+    /// Transparent wrapper around `std::string::FromUtf8Error`
+    FromUtf8(#[from] std::string::FromUtf8Error),
```

Comment:
I think it would be better to create a dedicated error for the Config parsing errors and ensure the `try_from` defined inside of this PR always returns this type of error.